core-bundle/src/EventListener/DataContainer/Page/AdjustAutoforwardSubpaletteListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer\Page;
```

Comment:
Is it really necessary to create a sub-namespace here?

Reply:
If, in the future, we implement more and more callbacks as services for different tables, then I think it would be best to organize it this way, yes. However, if we don't consider these services as part of the API, then we can simply reorganize it later in the future of course.

Reply:
I'm pro sub-namespace. This also won't cause any confusion should we for instance want to add a `Page` listener (or other things with the same name as the DCAs).

Reply:
>  if we don't consider these services as part of the API

Event listeners are not part of our BC promise (see https://docs.contao.org/dev/internals/bc-promise/). You should therefore add an `@internal` tag.

@contao/developers What is your opinion on the sub-namespaces?

Reply:
While reviewing #1382, I have noticed that we already have a namespace for DCA event listeners: `src/EventListeners/DataContainer`